grakn-graph/src/test/java/ai/grakn/graph/property/RelationTypePropertyTest.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.graph.property;
+
+import ai.grakn.GraknGraph;
+import ai.grakn.concept.Relation;
+import ai.grakn.concept.RelationType;
+import ai.grakn.concept.RoleType;
+import ai.grakn.exception.ConceptException;
+import ai.grakn.generator.AbstractTypeGenerator.Meta;
+import ai.grakn.generator.FromGraphGenerator.FromGraph;
+import ai.grakn.generator.GraknGraphs.Open;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.pholser.junit.quickcheck.Property;
+import com.pholser.junit.quickcheck.runner.JUnitQuickcheck;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.util.Set;
+
+import static ai.grakn.graph.property.PropertyUtil.choose;
+import static ai.grakn.graph.property.PropertyUtil.directSubTypes;
+import static ai.grakn.util.ErrorMessage.META_TYPE_IMMUTABLE;
+import static ai.grakn.util.ErrorMessage.SUPER_TYPE_LOOP_DETECTED;
+import static ai.grakn.util.Schema.MetaSchema.isMetaName;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * @author Felix Chapman
+ */
+@RunWith(JUnitQuickcheck.class)
+public class RelationTypePropertyTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Property
+    public void whenDeletingMetaRelationType_Throw(@Open GraknGraph graph) {
+        RelationType relation = graph.admin().getMetaRelationType();
+
+        exception.expect(ConceptException.class);
+        exception.expectMessage(META_TYPE_IMMUTABLE.getMessage(relation.getName()));
+
+        relation.delete();
+    }
+
+    @Property
+    public void whenANonMetaRelationTypeHasNoInstancesSubTypesOrRules_ItCanBeDeleted(
```

Comment:
The reason this exists is because some types has different rules about when they can be deleted. I already have a "general" deletion test in `ConceptPropertyTest` that I intend to remove eventually - the preconditions for deletion are impossible to understand - look at `assumeDeletable` to see for yourself. The test is actually complicated enough that it could easily be incorrect and I wouldn't know...

Let me know if you have any ideas on this front, though! I don't like duplicating tests either.